src/ui/public/kuery/ast/kql.peg

```diff
@@ -1,123 +1,147 @@
-/*
- * Kuery parser
- */
-
-/*
- * Initialization block
- */
+// Initialization block
 {
-  var nodeTypes = options.helpers.nodeTypes;
+  var { nodeTypes: { function: { buildNode }}} = options.helpers;
+}
+
+start
+  = OrQuery
+  / EmptyQuery
 
-  if (options.includeMetadata === undefined) {
-    options.includeMetadata = true;
+EmptyQuery
+  = Space* {
+    return buildNode('and', []);
   }
 
-  function addMeta(source, text, location) {
-    if (options.includeMetadata) {
-      return Object.assign(
-        {},
-        source,
-        {
-          text: text,
-          location: simpleLocation(location),
-        }
-      );
-    }
+OrQuery
+  = left:AndQuery Or right:OrQuery {
+    return buildNode('or', [left, right]);
+  }
+  / AndQuery
 
-    return source;
+AndQuery
+  = left:NotQuery And right:AndQuery{
+    return buildNode('and', [left, right]);
   }
+  / NotQuery
 
-  function simpleLocation(location) {
-    // Returns an object representing the position of the function within the expression,
-    // demarcated by the position of its first character and last character. We calculate these values
-    // using the offset because the expression could span multiple lines, and we don't want to deal
-    // with column and line values.
-    return {
-      min: location.start.offset,
-      max: location.end.offset
+NotQuery
+  = Not query:SubQuery {
+    return buildNode('not', query);
+  }
+  / SubQuery
+
+SubQuery
+  = '(' Space* query:OrQuery Space* ')' { return query; }
+  / Expression
+
+Expression
+  = FieldListExpression
+  / FieldRangeExpression
+  / ValueExpression
+
+FieldListExpression
+  = field:Literal Space* ':' Space* list:SubList {
+    function buildNodeWithField(node) {
+      if (node.hasOwnProperty('arguments')) {
+        const args = node.arguments.map(buildNodeWithField);
+        return { ...list, arguments: args };
+      } else {
+        return buildNode('is', field, node);
+      }
     }
+    return buildNodeWithField(list);
   }
-}
 
-start
-  = Query
-  / space* {
-    return addMeta(nodeTypes.function.buildNode('and', []), text(), location());
+FieldRangeExpression
+  = field:Literal Space* operator:RangeOperator Space* value:Literal {
+    return buildNode('range', field, { [operator]: value });
   }
 
-Query
-  = space? query:OrQuery space? {
-    if (query.type === 'literal') {
-      return addMeta(nodeTypes.function.buildNode('and', [query]), text(), location());
-    }
-    return query;
+ValueExpression
+  = value:Literal {
+    return buildNode('is', '*', value);
```

Comment:
Just noticed this when I was working on wildcards. We should pass `null` instead of `*` as the second argument here. `*` will force an all fields query, `null` will fall back on the default field set in the index settings, which Beats requires since the number of fields in their index patterns max out the new limit on # of fields in a single query in ES.